service/gsheet_bot.py

```diff
@@ -0,0 +1,210 @@
+    df = get_all_user(worksheet)
+
+    if df.empty:
+        logger.info("No data found. Skipping update.")
+        return True
+
+    df_pass = df[df['status'] == 'pass']
+
+    if df_pass.empty:
+        logger.info("No pass user to be updated.")
+        return True
+
+    update_scripts = []
+    insert_scripts = []
+    free_vip_discord_ids = []
+
+    for _, row in df_pass.iterrows():
+        uuid = row['uuid']
+        discord_id = row['discord_id']
+        volume = row['volume']
+        status = row['status']
+
+        if not pd.notna(uuid) or not pd.notna(discord_id):
+            logger.info("Empty UUID or Discord ID found. Skipping entry.")
+            continue
+
+        free_vip_discord_ids.append(discord_id)
+
+        # Group all insert/update trade volume table
+        if dbcon.fetch_user_trade_volume_by_discord_id(discord_id):
+            update_scripts.append((discord_id, volume))
+        else:
+            insert_scripts.append((uuid, discord_id, volume))
+
+    execute_updates_and_inserts(dbcon, update_scripts, insert_scripts)
+    await assign_free_vip(guild, free_vip_discord_ids)
+    return True
+
+
+async def batch_revoke_free_vip(dbcon, guild):
+    worksheet = get_worksheet(config.GBOT_GSHEET_ID)
+    df = get_all_user(worksheet)
+
+    if df.empty:
+        logger.info("No data found. Skipping update.")
+        return True
+
+    df_fail = df[df['status'] == 'fail']
+
+    if df_fail.empty:
+        logger.info("No fail user to be updated.")
+        return True
+
+    update_scripts = []
+    insert_scripts = []
+    revoke_free_vip_discord_ids = []
+
+    for _, row in df_fail.iterrows():
+        uuid = row['uuid']
+        discord_id = row['discord_id']
+        volume = row['volume']
+        status = row['status']
+
+        if not pd.notna(uuid) or not pd.notna(discord_id):
+            logger.info("Empty UUID or Discord ID found. Skipping entry.")
+            continue
+
+        revoke_free_vip_discord_ids.append(discord_id)
+
+        # Group all insert/update trade volume table
+        if dbcon.fetch_user_trade_volume_by_discord_id(discord_id):
+            update_scripts.append((discord_id, volume))
+        else:
+            insert_scripts.append((uuid, discord_id, volume))
+
+    execute_updates_and_inserts(dbcon, update_scripts, insert_scripts)
+    await revoke_free_vip_role(guild, revoke_free_vip_discord_ids)
+    return True
+
+
+def execute_updates_and_inserts(dbcon, update_scripts, insert_scripts):
+    if update_scripts:
+        update_cases_volume = " ".join(
+            f"WHEN '{discord_id}' THEN '{volume}'"
+            for discord_id, volume in update_scripts
+        )
+        update_discord_ids = ", ".join(f"'{discord_id}'" for discord_id, _ in update_scripts)
+
+        try:
+            dbcon.update_trade_volume_table_by_discord_id(update_cases_volume, update_discord_ids)
+            logger.info(f"{len(update_scripts)} records updated at trade_volume_table.")
```

Comment:
remove this, this will flood the log

Reply:
logger.info(f"{len(update_scripts)} records updated at trade_volume_table.")

this just log how many rows updated in the db, no loop 1000 rows to write in log